project_stock/models/project_project.py

```diff
@@ -31,7 +31,25 @@ class ProjectProject(models.Model):
         help="Default location to which materials are consumed.",
     )
 
+    def _check_tasks_with_pending_moves(self):
+        self.task_ids._check_tasks_with_pending_moves()
+
     @api.onchange("picking_type_id")
-    def onchange_picking_type(self):
+    def onchange_picking_type_id(self):
         self.location_id = self.picking_type_id.default_location_src_id.id
         self.location_dest_id = self.picking_type_id.default_location_dest_id.id
+
+    @api.onchange("location_id")
+    def onchange_location_id(self):
```

Comment:
Use convention and avoid that this can be publicly called (the same with the rest):

```suggestion
    def _onchange_location_id(self):
```

Anyways, I'm thinking this should be done only on write, not on onchange, as anyways it should be checked again on that final step.